providers/azureblob/src/main/java/org/jclouds/azureblob/blobstore/AzureBlobRequestSigner.java

```diff
@@ -105,9 +111,28 @@ public HttpRequest signGetBlob(String container, String name, org.jclouds.blobst
       return sign("GET", container, name, blob2HttpGetOptions.apply(checkNotNull(options, "options")),
             DEFAULT_EXPIRY_SECONDS, null, null);
    }
-
-   private HttpRequest sign(String method, String container, String name, @Nullable GetOptions options, long expires, @Nullable Long contentLength, @Nullable String contentType) {
-      checkNotNull(method, "method");
+   
+   /**
+   * method to parse SAS string into components like st, se, sr etc.
+   *
+   */
+   public HashMap<String, String> refineSAS(String sas) {
+		HashMap<String, String> paramsMap = new HashMap<>();
```

Comment:
jclouds prefers `ImmutableMap` for instances and `Map` for interfaces, e.g., return types.